lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -167,26 +181,27 @@
     free_text_fields=["user", "tool", "handler", "runner"],
 )
 
+FullShowQueryParam: Optional[bool] = Query(title="Full show", description="Show extra information.")
+HdaLddaQueryParam: DatasetSourceType = Query(
+    title="HDA or LDDA", description="Whether this dataset belongs to a history (HDA) or a library (LDDA)."
+)
+
+
+JobIdPathParam: DecodedDatabaseIdField = Path(title="Job ID", description="The ID of the job")
+DatasetIdPathParam: DecodedDatabaseIdField = Path(title="Dataset ID", description="The ID of the dataset")
+
+ReportErrorBody = Body(default=Required, title="Report error", description="The values to report an Error")
+SearchJobBody = Body(default=Required, title="Search job", description="The values to search an Job")
+# TODO The endpoint only stops/cancels a job, but is called delete settle for one name
+DeleteJobBody = Body(title="Delete/cancel job", description="The values to delete/cancel a job")
+
 
 @router.cbv
 class FastAPIJobs:
     service: JobsService = depends(JobsService)
-
-    @router.get("/api/jobs/{id}")
-    def show(
-        self,
-        id: DecodedDatabaseIdField,
-        trans: ProvidesUserContext = DependsOnTrans,
-        full: Optional[bool] = False,
-    ) -> Dict[str, Any]:
-        """
-        Return dictionary containing description of job data
-
-        Parameters
-        - id: ID of job to return
-        - full: Return extra information ?
-        """
-        return self.service.show(trans, id, bool(full))
+    manager: JobManager = depends(JobManager)
+    hda_manager: HDAManager = depends(HDAManager)
+    job_search: JobSearch = depends(JobSearch)
```

Comment:
You already have the `JobManager`, `HDAManager`, and `JobSearch` instances in your `JobService`.

In general, I recommend encapsulating the logic of the routes in the `JobService` so the API module only contains FastAPI documentation and basic parameter or response massaging. By keeping the logic in a function inside the service we can easily change in the future to a different API framework more cleanly.

Reply:
Thank you for reminding me :) I adjusted it